### ISSUE TITLE
Next Release

### ISSUE THREAD

**Issue by digitaldan** [openhab/openhab-android#549]
This issue is used to coordinate new releases.

**Comment by lolodomo** [comment]
The translations have to be emhanced too as some texts were added without translation just before the release 2.0.

**Comment by FlorianSW** [comment]
#561 should be included, too.

**Comment by digitaldan** [comment]
Hi guys, was away for the holidays and other engagements, but I'm back now.

**Comment by cribskip** [comment]
Just a little follow-up: I've deployed a build of the [current master](https://github.com/openhab/openhab-android/tree/e6466c5d7debe947558db4608e9788e665ddb265) to my phone and tested the enhancements to this scope:

- #628 Voice widget
- #517 NFC

btw, these are the only ones merged from the list above

**Comment by mueller-ma** [comment]
@digitaldan I would suggest to release a new version soon, because 2.0.0 crashes on all kitkat devices, which is about ~15% of all users.

**Comment by mueller-ma** [comment]
2.0.24-beta (rename to .apk before): [mobile-full-debug.pdf](https://github.com/openhab/openhab-android/files/1799296/mobile-full-debug.pdf)

**Comment by mueller-ma** [comment]
For now I have found following changes that should be in the play store changelog:

```
1. Allow the widget list to be refreshed when swiping (#639)
2. Fix flickering icons on sitemap reload. (#669)
3. Improve look'n'feel of section switch button (#687)
4. Improved auto discovery (#562)
5. Changed visual look of color and rollershutter buttons (#736)
6. New error fragment (#688) 
```

I think we can merge some of those:

```
1. Allow the widget list to be refreshed when swiping
3+5. Improve look'n'feel of some items
6. <removed>
2+4. Stability and performance improvements
```

**Comment by maniac103** [comment]
6. + the navigation drawer overhaul can be summarized as something like 'UI improvements'

**Comment by mueller-ma** [comment]
And can be merged with 3+5:
```
1. Allow the widget list to be refreshed when swiping
3+5+6. UI improvements
2+4. Stability and performance improvements
```

**Comment by mueller-ma** [comment]
Updated changelog:
```
Allow the widget list to be refreshed when swiping
Support for Mapview widgets
Adapt voice service to use new voice command endpoint
UI improvements
Stability and performance improvements
```

**Comment by mueller-ma** [comment]
#830, #817 and maybe #816 and than it should be fine for 2.1.0

**Comment by maniac103** [comment]
Should add #828 as well.

**Comment by mueller-ma** [comment]
German changelog:
```
Erlaube aktualisieren der Sitemap durch nach-unten-wischen
Unterstützung für Mapview-Elemente
Sprachbefehlfunktion an neuen Endpunkt angepasst
UI-Verbesserungen
Stabilitäts- und Performance-Verbesserungen
```
@maniac103 Does this look ok for you?

**Comment by maniac103** [comment]
I'd use `Sitemap-Aktualisierung durch Wischen nach unten` and `Unterstützung für Sprachbefehls-API des Servers, falls verfügbar`

**Comment by mueller-ma** [comment]
Thanks. New version is live.

**Comment by mueller-ma** [comment]
Let's use this issue to track all releases.
I plan to release 2.2.0 before the 2018-05-25, because of the GDPR (#878). It will have the codebase of 2.1.9-beta plus updated translations (#883).
Changelog (en):
````
Implement a preference controlling chart scaling
Stability and performance improvements
````
de:
````
Eine Einstellung wurde hinzugefügt, um die Skalierung der Diagramme anzupassen
Stabilitäts- und Performance-Verbesserungen
````

**Comment by maniac103** [comment]
`Eine Einstellung wurde hinzugefügt, um die Skalierung der Diagramme anzupassen` -> `Einstellbare Diagrammskalierung`?

**Comment by mueller-ma** [comment]
2.2.0 is released.

**Comment by mueller-ma** [comment]
It has been some months since the last so we can release 2.3.0 in the near future. However those two bugs seem to be a deal-breaker for me: #953 and #955 
#953 could be "fixed" by changing targetApi to 24 for now.

**Comment by mueller-ma** [comment]
Poking translator, so they are aware of this release ticket.
@WhiteOwl @FrankCZ @gargamelonly @gudszent @hakan42 @Confectrician @kubawolanin @martinvw @gitMiguel @ghys @4digital @ahielg @ahmed89albttashi @alicangec @velmyshanovnyi @asero @barzah @bonieP @brookstudio

**Comment by maniac103** [comment]
> It has been some months since the last so we can release 2.3.0 in the near future. However those two bugs seem to be a deal-breaker for me: #953 and #955

I'll have a look at both in the next week-ish.

**Comment by mueller-ma** [comment]
With https://github.com/openhab/openhab-android/pull/980 probably being the last PR that will be merged for 2.3.0 you can start working on the translations.

**Comment by mueller-ma** [comment]
Anything against a release this week?

**Comment by maniac103** [comment]
I'd suggest looking at/merging #988 before, otherwise fine with me.

**Comment by mueller-ma** [comment]
Notable changes in 2.3:

1. MTM (we should add this to the release notes to avoid confusion why the mtm is showing up)
2. Formatting label text output #233 (Might be combined with 3, 4 and 5 as "UI improvements")
3. Add vibration and led light to notifications #920 (Should be combined with 5)
4. Use upstream HoloColorPicker instead of rolling our own. #917
5. Improve notification UI #848 

Suggestion:

• Enhanced security. You might get asked again to accept self-singed certificates.
• UI improvements (App and notifications)

**Comment by mueller-ma** [comment]
@maniac103 Any idea how [this crash](https://console.firebase.google.com/project/openhab-mobile/crashlytics/app/android:org.openhab.habdroid/issues/5bab4eabf8b88c29635a62d9) happened? The app has this permission.

**Comment by maniac103** [comment]
@mueller-ma Not sure, can't think of anything besides root hackery (system allowing permission denial of every permission, not only runtime ones), but Firebase says no root for this user.

**Comment by mueller-ma** [comment]
I justed tagged https://github.com/openhab/openhab-android/releases/tag/2.3.9-beta, this will be the latest beta before 2.4 as the distribution of 2.3 has been stopped due to a major bug.

**Comment by mueller-ma** [comment]
Changelog for 2.4 (includes changes from 2.3):
• Enhanced security. You might get asked again to accept self-singed certificates.
• UI improvements (App and notifications)
• Added setting to improve chart loading times
• Improve item updates

German:
• Erweiterte Sicherheit. Sie werden vielleicht erneut gefragt selbst signierte Zertifikate zu akzeptieren.
• UI Verbesserungen (App und Benachrichtigungen)
• Einstellung hinzugefügt, um die Ladezeiten von Diagrammen zu verkürzen
• Item-Aktualisierungen verbessert

**Comment by mueller-ma** [comment]
I want to release a new stable release soon to adress some issues in 2.4.0:
• Fix issue with Grafana graphs
• Hide notification sound settings on Android 8 and above. Please use Notification channels to change it.
• Stability improvements

German:
• Problem mit Grafana Diagrammen behoben
• Verberge Benachrichtigungseinstellung auf Android 8 und höher. Bitte nutzen Sie die Benachrichtigungskanäle um den Benachrichtigungston zu ändern.
• Stabilitätsverbesserungen

**Comment by maniac103** [comment]
Improved German: ;-)

• Problem mit Grafana-Diagrammen behoben
• Benachrichtigungseinstellungen wurden auf Android 8 und höher. Bitte nutzen Sie stattdessen die Einstellungen der Benachrichtigungskanäle, um den Benachrichtigungston zu ändern.
• Stabilitätsverbesserungen

**Comment by mueller-ma** [comment]
"• Benachrichtigungseinstellungen wurden auf Android 8 und höher" verborgen?

**Comment by maniac103** [comment]
Oops, yeah, right :+1:

**Comment by mueller-ma** [comment]
Changelog for upcomming 2.6:
• Add app shortcuts
• Add ability to view HABPanel
• Fix issues with units, visibility and webview widgets

• App Shortcuts hinzugefügt
• HABPanel-Anzeige hinzugefügt
• Probleme mit Einheiten, Sichtbarkeit und Webview-Elementen behoben

I'll make #1167 ready for 2.6, but #1174 and #1137 should go to 2.7.

**Comment by mueller-ma** [comment]
@maniac103 IMO after merging #1227 and the latest translations I can release 2.6.0. Thoughts?

**Comment by maniac103** [comment]
@mueller-ma Sounds fine to me. I'm not aware of any other urgent things at the moment.
BTW, myopenhab will probably get updated this weekend with severity/icon support for notifications. I wonder whether we should make people aware of that somehow?

**Comment by mueller-ma** [comment]
I want to test #1226 and check if everything works fine (at least on Android 7.1) with the GCM server component. If anything with the server-side migration will go wrong and the GCM client doesn't work anymore, we can merge #1226 and release a new stable version very quickly. In this case the release notes for 2.7.0 will contain a note that notification channels are now fully supported.
Or we ship 2.7.0 in any case soon and include some smaller PRs (#1211, #1218, maybe a fix for #1221).

**Comment by maniac103** [comment]
I'm reasonably sure there won't be an issue with that (FCM on server, GCM client) because I specifically tested that when rebasing my openhab-cloud patch. Making sure that FCM works on client side in a good idea nevertheless though :-)

**Comment by mueller-ma** [comment]
Due to crashes at app startup (#1235) I stopped the distribution of 2.6.0 and I want to release 2.6.X pretty soon.

**Comment by mueller-ma** [comment]
@maniac103 Anything against release 2.7.0 this week? Since all commits are mostly bugfixes, I would use the changelog from 2.6.0.

**Comment by maniac103** [comment]
@mueller-ma That sounds fine to me.

**Comment by mueller-ma** [comment]
Released as 2.6.9.

**Comment by cribskip** [comment]
the release reintroduced #1221 by sending UoM

**Comment by maniac103** [comment]
> the release reintroduced #1221 by sending UoM

#1250 is part of this release though ...

**Comment by mueller-ma** [comment]
> the release reintroduced #1221 by sending UoM

I couldn't reproduce #1221 with 2.6.9.

**Comment by cribskip** [comment]
@mueller-ma just tested 2.6.9-release vs. 2.6.7-beta. Issue is noticable in the events log:

2.6.7-beta, no UoM:
```
2019-03-15 18:21:34.135 [ome.event.ItemCommandEvent] - Item 'TSWohnen' received command 21.2
2019-03-15 18:21:34.163 [vent.ItemStateChangedEvent] - TSWohnen changed from 21.1 to 21.2
```

With 2.6.9-release, notice the "°C" UoM:
```
2019-03-15 18:21:51.782 [ome.event.ItemCommandEvent] - Item 'TSWohnen' received command 21.1 °C
2019-03-15 18:21:51.813 [vent.ItemStateChangedEvent] - TSWohnen changed from 21.2 to 21.1 °C
```

**Comment by cribskip** [comment]
I'm stupid, still had 2.6.0-release installed while noticing the bug. Thought to have upgraded.

Just verified working 👍

**Comment by mueller-ma** [comment]
I think it's time for a new release before #1280 and #1302 are merged. #1328 should be fixed before releasing 2.7.0. So far we have:
````
• Send alarm clock time to openHAB server
• Add up and down buttons to Setpoint widgets
• Clicking the widget row now toggles some widgets
````

**Comment by mueller-ma** [comment]
Anything against the next release this weekend?

**Comment by maniac103** [comment]
I need to fix the location parsing sanity check, but nothing besides that.

**Comment by mueller-ma** [comment]
#1361 should be merged as well.

**Comment by mueller-ma** [comment]
German changelog:

````
• Sende Weckzeit an openHAB server
• Hoch/Runter Buttons für Setpoint hinzugefügt
````

No idea how to translate the third one in a good way.

**Comment by mueller-ma** [comment]
I started the rollout of https://github.com/openhab/openhab-android/releases/tag/2.7.0-release for the first 20% of users.

**Comment by mueller-ma** [comment]
Translators, can you check if the following issues need to be fixed:
![grafik](https://user-images.githubusercontent.com/22525368/60399445-b065b080-9b64-11e9-909e-df0636f6b9bc.png)

**Comment by mueller-ma** [comment]
After merging #1424 and #1390 and fixing #1426, #1427, #1428, #1429, #1430, we should be ready for a new stable release.

**Comment by mueller-ma** [comment]
I don't see any major issues or crashes in the latest beta, so after #1455 version 2.8.0 is ready.
````
• Add Tasker/Locale action plugin
• Add ability to create sitemap shortcuts
• Fix issues with visibility flag
• Stability and performance improvements
````

````
• Tasker/Locale Action Plugin hinzugefügt
• Sitemap-Verknüpfungen können auf Startbildschirm erstellt werden
• Fehler mit "visibility"-Flag behoben
• Stabilitäts- und Performance-Verbesserungen
````

**Comment by mueller-ma** [comment]
Version 2.8.0 is currently rolled out to 50%.

**Comment by mueller-ma** [comment]
Google rejected the update to 2.8.0. I'll check why.

**Comment by mueller-ma** [comment]
2.8.0 is live again.

**Comment by mueller-ma** [comment]
Changelog for 2.9.0. I don't see any severe issues with the current beta.

````
• Send call state to server
• Add widgets
• Add lock screen
• Improve Video widgets
• Needs Android 4.4 KitKat or higher
````

````
• Sende Anrufstatus an den Server
• Widgets hinzugefügt
• Bildschirmsperre hinzugefügt
• Video-Widgets verbessert
• Benötigt Android 4.4 KitKat oder höher
````

**Comment by mueller-ma** [comment]
I suggest to release 2.10.0 soon, after resolving #1602.
```
• Write NFC tags for Item
• Support Android 10 dark mode
• Fix lock screen on Android 10
```

```
• Schreibe NFC-Tags für Items
• Dunkler Modus in Android 10 wird unterstützt
• Probleme mit Sperrbildschirm auf Android 10 behoben
```

After that release we can merge #1600, which requires some testing.

**Comment by mueller-ma** [comment]
IMO no new issue is critical. Let's give the translators a few days time and than release 2.10.0.
@maniac103 WDYT?

**Comment by maniac103** [comment]
@mueller-ma :+1:

**Comment by mueller-ma** [comment]
IMO #1645 and #1632 are severe enough for 2.10.1-release.

**Comment by maniac103** [comment]
#1643 should be put into that release as well since it's a regression.

**Comment by mueller-ma** [comment]
I released 2.10.2-beta. Let's wait a day and see if any new crashes appear. If none appear, merge the latest translations and release 2.10.3-release.
After that, merge #1648 and #1594 (If reviewed and working). I guess the two authors participate in Hacktoberfest.
2.10.0-release is still at 20% distribution in the play store.

**Comment by mueller-ma** [comment]
I added 7 new translation of the description for both the stable version and the beta:

- Dutch
- Finnish
- Japanese 
- Norwegian 
- Polish 
- Romanian 
- Ukrainian 

Sadly, I cannot add Luxembourgish, because it's not available in the Play Store.

**Comment by mueller-ma** [comment]
Changelog for 2.11.0. I don't see any severe issues with the current beta.

````
• Improve widgets and Tasker plugin.
• Show only the default Sitemap in the side menu. There's a new setting to show all Sitemaps there.
• Disable http for new remote connections.
````

````
• Widgets und Tasker-Plugin verbessert.
• Zeige nur die Standard-Sitemap im seitlichen Menü. Es gibt eine neue Einstellung, um alle anzuzeigen.
• http für neue Fernzugriffsverbindungen deaktiviert.
````

**Comment by mueller-ma** [comment]
After merging #1729 the new release should be fine. I want to postpone #1719 (so 2.11.0 will be available for Android 4) and #1720 and #1722 (should be in beta for some time to catch some edge cases).

**Comment by mueller-ma** [comment]
What we have so far (2.11.21-beta):
````
• Click on charts to view them full screen
• Send UNDEF instead of 0 if no alarm is active
• Add ability to create shortcut with dark background
````

**Comment by mueller-ma** [comment]
After merging #1854 and updating the translations, 2.12.0 should be fine.

**Comment by maniac103** [comment]
> Vollbildanzeige für Diagramme mittels Klick auf das Diagramm
> Wenn kein Alarm aktiv ist, wird nun UNDEF anstatt 0 an das Item gesendet

(Der Imperativ passt aus meiner Sicht nicht so gut, wenn einmal der Nutzer und das andere Mal die App die Tätigkeit ausführen)

**Comment by mueller-ma** [comment]
After merging #1860 I released and uploaded a new beta version. If there are no new issues, I'll release it as stable version.
Due to Covid-19, the Play Store reviews may take up to 7 days.

**Comment by mueller-ma** [comment]
> Due to Covid-19 the Play Store reviews may take up to 7 days.

It's live.

**Comment by mueller-ma** [comment]
For 2.13.0 we have so far:
````
• Send Wi-Fi name, battery level and charging state to server
• Support system-wide data saver setting and add a setting to turn it on app-wide
• New color picker for Color items
````

**Comment by lolodomo** [comment]
No chance to include a fix for #1891 in 2.13.0 ?

**Comment by mueller-ma** [comment]
Sure, I edited my comment above.

**Comment by mueller-ma** [comment]
https://github.com/openhab/openhab-android/releases/tag/2.12.21-beta should be fine to release as 2.13.0. I'll wait a few days for translators and maybe new crashes will show up.

**Comment by mueller-ma** [comment]
I tagged https://github.com/openhab/openhab-android/releases/tag/2.13.0-release and I'll start distribution in Play Store at 20% later today.

**Comment by mueller-ma** [comment]
@maniac103 Do you think it's worth to tag a new release version with the three commits on master (#1992, #2037, #2038)? Otherwise I set the current 2.13.0 to 100% distribution.

**Comment by maniac103** [comment]
@mueller-ma It surely doesn't hurt, but I don't think it's really necessary. All 3 are pretty minor things hardly any user will notice.

**Comment by mueller-ma** [comment]
Ok, I'm the same opinion. 2.13.0 is now at 100%.

**Comment by mueller-ma** [comment]
For 2.14.0 we have so far:
````
• Control Items from the quick settings of your device (the place that holds the toggles for e.g. Wi-Fi and Bluetooth)
• Add a new setting: Device identifier. Please have a look at the setting itself, as it's quite self explanatory.
• Support notifications in the FOSS flavor 
````

The last one won't be displayed in the Play Store.

#2072 should be tested for some time in beta before releasing 2.14.0.

**Comment by mueller-ma** [comment]
@maniac103 What do you think about a new stable release? Currently the app is quite stable, so a release should be fine.

**Comment by maniac103** [comment]
Yeah, makes sense to me as well.

**Comment by cwihne** [comment]
Still having issues with video url crashing the app, now with 2.13.20. Yet, little more fuzzy, not reproducible with simple sitemap, trying to provide solid log info.

**Comment by cwihne** [comment]
commented with log in https://github.com/openhab/openhab-android/issues/1802

**Comment by mueller-ma** [comment]
@cwihne Thanks for testing, but this issue is also present in the stable 2.13.0 release.

**Comment by cwihne** [comment]
@mueller-ma OK, so why holding up new stable release any longer? ;-)

**Comment by mueller-ma** [comment]
The translations need to be updated and I'd like to keep the current beta version for a few more days in beta.

**Comment by mueller-ma** [comment]
I started the rollout of 2.14.0 at 20%.

**Comment by mueller-ma** [comment]
I'd like to release 2.15.0 soon to get rid of all the video crashes:
````
• Fix crash when using video Items
• Add long press behavior for the up/down buttons on roller shutter Items
• Stability and performance improvements
````

It wouldn't hurt to include #2172, #2153 and #2164 as well.

**Comment by mueller-ma** [comment]
2.15.0 is now at 20%.

**Comment by mueller-ma** [comment]
Milestone for 2.16.0: https://github.com/openhab/openhab-android/milestone/15
#2140 is a must for 2.16.0, the others are all optional. I'd like to keep it only for a short time in beta, because Android 11 is already released.

**Comment by maniac103** [comment]
> I'd like to keep it only for a short time in beta, because Android 11 is already released.

It'll be some time before targeting API 30 will become mandatory, though.

**Comment by mueller-ma** [comment]
Except the hint that users shouldn't select "only this time" when asked for location permissions, the improvements in #2140 only apply for apps targeting Android 11.

**Comment by mueller-ma** [comment]
What we have for 2.16.0 so far:
````
• Add support for configuring multiple openHAB servers
• Add support for Android 11
````

I'd like to see if https://github.com/openhab/openhab-android/releases/tag/2.15.10-beta fixes all snackbar related crashes first.

**Comment by mueller-ma** [comment]
If no new bugs in 2.15.27-beta appear, it should be fine to release 2.16.0 soon.

**Comment by mueller-ma** [comment]
What we have for 2.17.0 so far:
````
• Improve widgets
• Add support for configuring start page (Sitemaps, HABPanel, etc.)
• Click on images to view them full screen
• Add support for Frontail
````

Currently I don't see any major issues with the latest beta version.

**Comment by mueller-ma** [comment]
Proposal for 2.18.0:
````
• Updated translations
• Stability and performance improvements
````

**Comment by mueller-ma** [comment]
2.20.0 can be released after fixing #2762:

````
• Add support for Android 11 device controls 
• Render section switch as selection when too wide 
````

**Comment by mueller-ma** [comment]
Proposal for 2.21.0:
````
• Show item state in home screen widget 
• Send list of connected Bluetooth devices
• Improve support for Android's device control
• Add ability to restrict local connection to specific Wi-Fis
• Better debugging of sending alarm time. You might get more notifications when turning on debug mode now.
````
